projects/meta_cl/experiment_classes.py

```diff
@@ -0,0 +1,43 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from nupic.research.frameworks.vernon import MetaContinualLearningExperiment, mixins
+
+
+class OMLExperiment(mixins.OnlineMetaLearning,
+                    MetaContinualLearningExperiment):
+    pass
+
+
+class IIDTrainingOMLExperiment(OMLExperiment):
+    def sample_slow_data(self, tasks):
```

Comment:
Isn't this just removing the replay set, instead of replacing for one that is sampled from all existing classes? Is that a fair comparison?

You also argued the difference for the iid setting is that in the inner loop we are no longer seeing 20 images from a single class, but 20 images sampled from any class. Is that change also part of this PR?

Reply:
Per each question:
1) The replay set is still sampled. In fact, it's the only data that would now get used in the outer loop. For instance, if there were 30 images before, 15 from the replay set, and 15 sampled from the task(s), then now there would be only 15 from the replay set.

2) The inner loop was already capable of this. You do this by setting `tasks_per_epoch=20` and `num_fast_steps=1`. The tasks are sampled randomly and you take `num_fast_steps` per task. Thus, you take one gradient step over one image sampled accross 20 random tasks, as you request.